apps/api/src/domain/models/song/song.entity.ts

```diff
@@ -142,6 +160,75 @@ export class Song extends Resource implements ITimeBoundable {
         return allErrors.concat(titleValidationErrors);
     }
 
+    // TODO consider making this any iterable of BaseEvents
```

Comment:
See if you can do this now.

Reply:
It's tricky and not currently worth the effort. Only if we think we want to use a map or some other data structure for arrays should we pursue this.